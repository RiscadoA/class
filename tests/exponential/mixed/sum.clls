proc main() {
    cut {
        println("!"); !a(b: choice of { |#C: close |#W: wait });
        println("#A"); #W b;
        println("wait"); wait b;
        ()
        |a: ? offer of { |#C: wait |#W: close }|
        println("?"); ?a;
        println("call 1"); call a(y);
        println("case of 1"); case y of {
            |#C:
                println("never"); wait y;
                ()
            |#W:
                par {
                    println("close 1"); close y
                    ||
                    println("call 2"); call a(z);
                    println("case of 2"); case z of {
                        |#C:
                            println("never"); wait z;
                            ()
                        |#W:
                            println("close 2"); close z
                    }
                }

        }
    }
};;
