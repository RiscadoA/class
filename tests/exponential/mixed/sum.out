!
?
call 1
#A
case of 1
close 1
wait
call 2
#A
case of 2
close 2
wait
